### PR TITLE
Initialise repo

### PR DESCRIPTION
**Pull request by aclevername**
This PR setups the repo for development
- PR/Issue templates
- Github actions for testing/lint/label enforcing
- Integration/unit tests (some redundant bare bone code added to ensure it works)

I followed the recommendations from  [golang-standards](https://github.com/golang-standards/project-layout) for the repo layout. 

I added a `version.go` with unit tests just so that some unit tests exists, it will probably be replaced/removed shortly

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+build:
+	go build -o ./pctl cmd/pctl/main.go
+
+test: unit-test integration-test
+
+unit-test:
+	go test -count=1 ./pkg/...
```
Comment (aclevername) [inline Makefile:7]: `-count=1` is the recommended way to disable test caching https://stackoverflow.com/questions/48882691/force-retesting-or-disable-test-caching

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Help your reviewers my guiding them through your key changes,
+implementation decisions etc.
+You can even include snippets of output or screenshots.
+
+A good, clear description == a faster review :)
+-->
+
+### Checklist
+- [ ] Added tests that cover your change (if possible)
```
Comment (Callisto13) [inline .github/PULL_REQUEST_TEMPLATE.md:14]:
```suggestion
- [ ] Added tests that cover your change
```

Testing is not optional any more

Comment (aclevername) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: :100:

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+implementation decisions etc.
+You can even include snippets of output or screenshots.
+
+A good, clear description == a faster review :)
+-->
+
+### Checklist
+- [ ] Added tests that cover your change
+- [ ] Added/modified documentation as required (such as the `README.md`)
+- [ ] Made sure the title of the PR is a good description that can go into the release notes
```
Comment (Callisto13) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: If we have the PR labels workflow, do we want to have the add labels check here?

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    - uses: actions/stale@v3
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-issue-message: 'This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
+        close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+        stale-issue-label: 'stale'
+        days-before-stale: 30
+        days-before-close: 5
+        start-date: '2018-06-01T00:00:00+00:00'
+        exempt-issue-labels: 'priority/critical,priority/important-longterm,priority/important-soon,priority/weave,good first issue,pinned'
```
Comment (Callisto13) [inline .github/workflows/stale.yml:20]: don't forget to add these labels to the tickets so we don't get burned 😂

File: pkg/cmd/version.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(versionCmd)
+}
+
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Output the version of PCTL",
```
Comment (Callisto13) [inline pkg/cmd/version.go:17]: PCTL all caps eh? bold choice

File: pkg/cmd/root.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
```
Comment (Callisto13) [inline pkg/cmd/root.go:7]: ugh cobra

**Comment by Callisto13** [comment]
Does `cmd` usually live in `pkg`? I usually see it live outside the main library collection

**Comment by Callisto13** [comment]
Do we want to rename this repo to `pctl` now? I called this `profiles` just to have something, but changing may make sense 🤷‍♀️

File: pkg/cmd/version.go (hunk 5 above)
Comment (aclevername) [reply pkg/cmd/version.go:17]: I stand by this.

**Comment by aclevername** [comment]
> Does `cmd` usually live in `pkg`? I usually see it live outside the main library collection

The `cmd` for main is outside `pkg/`, the `cmd` inside `pkg/` is for the commands following cobras recommendations (happy for us to deviate from)

File: Makefile (hunk 1 above)
Comment (bigkevmcd) [reply Makefile:7]: I'm curious about _why_ we're doing this?

Are we seeing issues that require disabling test caching?

**Comment by aclevername** [comment]
> Do we want to rename this repo to `pctl` now? I called this `profiles` just to have something, but changing may make sense

:man_shrugging: I think its cheap to change, until we have another component we can just keep it as is IMO

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (aclevername) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: done

File: pkg/cmd/version.go (hunk 5 above)
Comment (Callisto13) [reply pkg/cmd/version.go:17]: but... ew 😭

Comment (Callisto13) [reply pkg/cmd/version.go:17]: i feel like it is yelling at me

**Comment by Callisto13** [comment]
> The cmd for main is outside pkg/

ah yes i see that now

> the cmd inside pkg/ is for the commands following cobras recommendations (happy for us to deviate from)

ugh ofc gross cobra doing gross things

File: Makefile (hunk 1 above)
Comment (aclevername) [reply Makefile:7]: this is very much just personal preferences, I have found it unreliable in the past. I think it was that I altered an artefact outside of the test directory, so it didn't re-run the tests as it didn't detect a difference.

**Comment by aclevername** [comment]
@Callisto13 are you happy for us to use cobra? If you have an alternative in mind now is the best time for us to switch :grin:

**Comment by Callisto13** [comment]
I am a fan of https://github.com/urfave/cli

TBH the whole PCTL vs pctl upsets me more :trollface:

**Review by Callisto13 — APPROVED** [review]

**Pull request merged**